pkg/aws/eni/node.go

```diff
@@ -800,3 +794,29 @@ func (n *Node) getEffectiveIPLimits(eni *eniTypes.ENI, limits int) (effectiveLim
 	}
 	return effectiveLimits
 }
+
+// findSuitableSubnet attempts to find a subnet to allocate an ENI in according to the following heuristic.
+//  0. In general, the subnet has to be in the same VPC and match the availability zone of the
+//     node. If there are multiple candidates, we choose the subnet with the most addresses
+//     available.
+//  1. If we have explicit ID or tag constraints, chose a matching subnet. ID constraints take
+//     precedence.
+//  2. If we have no explicit constraints, try to use the subnet the first ENI of the node was
+//     created in, to avoid putting the ENI in a surprising subnet if possible.
```

Comment:
Actually, this is something we also want to explicitly mention in the upgrade guide (`upgrade.rst`).

Reply:
added a to `upgrade.rst` and `ipam/eni.rst`.